src/relations/db.py

```diff
@@ -270,6 +271,7 @@ def _on_relation_joined(self, join_event: RelationJoinedEvent):
             return
 
         # set up auth function
+        self.charm.backend.remove_auth_function(dbs=[database])
```

Comment:
If the auth_function was not removed, `create_database()` will change ownership and `initialise_auth_function()` will not recreate it.